app/src/main/java/systems/v/coldwallet/Util/UIUtil.java

```diff
@@ -408,10 +408,13 @@ public static void setExecContractTx(final Activity activity, final Account send
         String time = new Timestamp(timestamp).toString();
         time = time.substring(0, time.indexOf("."));
         timestampTx.setText(time + "\n" + TimeZone.getDefault().getDisplayName());
-
+        String tempAttachment = attachment;
         functionIdTx.setText(String.valueOf(functionId));
-
-        attachmentTx.setText(attachment);
+        try {
+            tempAttachment = new String(Base58.decode(attachment));
```

Comment:
We should not decode base58 here. The attachment will be decode again at Line 199 in `Transaction.java`. Decode twice will always trigger app crash.